Core/Frameworks/BaikalAdmin/Controller/Install/Database.php

```diff
@@ -226,4 +226,102 @@ function hideMySQLFieldWhenNeeded(\Formal\Form $oForm, \Formal\Form\Morphology $
+                    # Checking if all tables are missing
+                    $aRequiredTables = \Baikal\Core\Tools::getRequiredTablesList();
+                    if (count($aRequiredTables) !== count($aMissingTables)) {
+                        $sMessage = "<br /><p><strong>Database is not structurally complete.</strong></p>";
+                        $sMessage .= "<p>Missing tables are: <strong>" . implode("</strong>, <strong>", $aMissingTables) . "</strong></p>";
+                        $sMessage .= "<p>You will find the SQL definition of Baïkal tables in this file: <strong>Core/Resources/Db/PgSQL/db.sql</strong></p>";
+                        $sMessage .= "<br /><p>Nothing has been saved. <strong>Please, add these tables to the database before pursuing Baïkal initialization.</strong></p>";
+
+                        $oForm->declareError(
+                            $oMorpho->element("pgsql"),
+                            $sMessage
+                        );
+                    } else {
+                        # All tables are missing
+                        # We add these tables ourselves to the database, to initialize Baïkal
+                        $sSqlDefinition = file_get_contents(PROJECT_PATH_CORERESOURCES . "Db/PgSQL/db.sql");
+                        $oDb->getPDO()->exec($sSqlDefinition);
+                    }
+                }
+
+                return true;
+            } catch (\Exception $e) {
+                $oForm->declareError(
+                    $oMorpho->element("pgsql"),
+                    "Baïkal was not able to establish a connexion to the PostgreSQL database as configured.<br />PostgreSQL says: " . $e->getMessage()
+                );
+
+                $oForm->declareError(
+                    $oMorpho->element("pgsql_host")
+                );
+
+                $oForm->declareError(
+                    $oMorpho->element("pgsql_dbname")
+                );
+
+                $oForm->declareError(
+                    $oMorpho->element("pgsql_username")
+                );
+
+                $oForm->declareError(
+                    $oMorpho->element("pgsql_password")
+                );
+            }
+        }
+    }
+
+    public function validateSQLConnection($oForm, $oMorpho) {
+        if ($oMorpho->element("mysql")->value()) {
+            $this->validateMySQLConnection($oForm, $oMorpho);
+        } elseif ($oMorpho->element("pgsql")->value()) {
+            $this->validatePgSQLConnection($oForm, $oMorpho);
+        }
+    }
+
+    public function hideSqlFieldWhenNeeded(\Formal\Form $oForm, \Formal\Form\Morphology $oMorpho) {
+        if ($oMorpho->element("mysql")->value()) {
+            $this->hideMySQLFieldWhenNeeded($oForm, $oMorpho);
+        } elseif ($oMorpho->element("pgsql")->value()) {
+            $this->hidePgSQLFieldWhenNeeded($oForm, $oMorpho);
+        }
+    }
+
+    public function hidePgSQLFieldWhenNeeded(\Formal\Form $oForm, \Formal\Form\Morphology $oMorpho) {
```

Comment:
Something seems to be broken here. When using sqlite and visiting the settings page for the first time (without clicking any checkboxes), it shows the pgsql settings instead of the sqlite settings. After clicking the checkboxes it works as expected, so something is off with the default state.

Reply:
Oh dang, yeah that's quite a glitch. It should be fixed with the combined database backend dropdown menu introduced in 4ab75f76